ntbs-service/Services/ValidationService.cs

```diff
@@ -192,57 +196,64 @@ public ContentResult ValidateFullModel(object model)
                         keyErrorDictionary.TryAdd(modelStateKey, error.ErrorMessage);
                     }
                 }
-                return pageModel.Content(JsonConvert.SerializeObject(keyErrorDictionary));
+                return _pageModel.Content(JsonConvert.SerializeObject(keyErrorDictionary));
             }
         }
 
-        public ContentResult ValidateYearComparison(int yearToValidate, int yearToCompare, string propertyName)
+        public ContentResult GetYearComparisonValidationResult(int yearToValidate, int yearToCompare, string propertyName)
         {
             if (!IsValidYear(yearToValidate))
             {
-                return pageModel.Content(ValidationMessages.InvalidYear(propertyName));
+                return _pageModel.Content(ValidationMessages.InvalidYear(propertyName));
             }
 
             if (yearToValidate < yearToCompare)
             {
-                return pageModel.Content(ValidationMessages.ValidYearLaterThanBirthYear(propertyName, yearToCompare));
+                return _pageModel.Content(ValidationMessages.ValidYearLaterThanBirthYear(propertyName, yearToCompare));
             }
             else
             {
                 return ValidContent();
             }
         }
 
-        // We could do this validation using a custom data annotation, but as we already need to do another comparison
-        // it is simpler to do it here
-        private bool IsValidYear(int year)
+        public void ValidateProperty(object model, string modelKey, object modelProperty, string propertyKey)
```

Comment:
I'm not wholly sold on passing in all the key strings... but can't see a way around it without making too many assumptions about the state of the viewmodel.

Reply:
Yeah, I think you captured my feelings on it quite well :'D Afraid I've not got anything to suggest in this case..